packages/babel-parser/src/parser/expression.js

```diff
@@ -1919,6 +1994,43 @@ export default class ExpressionParser extends LValParser {
     return node;
   }
 
+  // parse an array literal or tuple literal
+  // https://tc39.es/ecma262/#prod-ArrayLiteral
+  // https://tc39.es/proposal-record-tuple/#prod-TupleLiteral
+  parseArrayLike(
+    close: TokenType,
+    isTuple: boolean,
+    canBePattern: boolean,
```

Comment:
Nit: `isTuple` and `canBePattern` are swapped in the `parseObjectLike` function. It might be good to also swap them here to avoid confusion.